compiler/qsc_frontend/src/resolve.rs

```diff
@@ -68,6 +68,10 @@ pub(super) enum Error {
     #[error("`{0}` not found")]
     #[diagnostic(code("Qsc.Resolve.NotFound"))]
     NotFound(String, #[label] Span),
+
+    #[error("This name shadows a global item.")]
+    #[diagnostic(code("Qsc.Resolve.ShadowsGlobal"))]
+    ShadowsGlobal(#[label] Span),
```

Comment:
I agree that this should be an error instead of a panic, but this isn't the right error message. There's nothing wrong with declaring something that shadows a global. And the error doesn't occur at the declaration, it occurs at the usage. This is a different flavor of `Error::Ambiguous`, where the ambiguity is between two or more namespaces *in the prelude* rather than two or more *explicitly opened* namespaces.